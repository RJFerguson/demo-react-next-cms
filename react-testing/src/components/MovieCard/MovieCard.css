.MovieCard {
  flex: 0 0 25%;
  border-radius: 25px;
  border: 2px solid #92a8d1;
  margin: 5px;
  padding-top: 5px;
}
