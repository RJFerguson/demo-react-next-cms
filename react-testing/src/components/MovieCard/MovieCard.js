import React from 'react';
import './MovieCard.css';
const MovieCard = ({ name, year, url }) => {
  return (
    <div className="MovieCard">
      <img alt="movies" src={`https://image.tmdb.org/t/p/w200/${url}`} />
      <div>
        <h3>{name}</h3>
        <p>{year}</p>
      </div>
    </div>
  );
};

export default MovieCard;
