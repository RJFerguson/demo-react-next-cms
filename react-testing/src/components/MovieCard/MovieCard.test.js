import React from 'react';
import { shallow, mount } from 'enzyme';
import MovieCard from './MovieCard';
import renderer from 'react-test-renderer';

describe('MovieCard Component', () => {
  // ------------- Unit testing ---------------
  it('renders without crashing', () => {
    const wrapper = shallow(<MovieCard />);
    expect(wrapper.length).toEqual(1);
  });

  it('renders a div with classname of MovieCard', () => {
    const wrapper = shallow(<MovieCard />);
    expect(wrapper.find('div.MovieCard')).toHaveLength(1);
  });

  it('renders a MovieCard with the correct name and year based on props passed', () => {
    const wrapper = shallow(<MovieCard name={'movie'} year={'2000'} url={'abc'} />);
    expect(wrapper.find('h3').text()).toEqual('movie');
    expect(wrapper.find('p').text()).toEqual('2000');
    expect(wrapper.find('img').prop('src')).toEqual('https://image.tmdb.org/t/p/w200/abc');
  });

  // -------------------- Snapshot Testing ------------------------
  it('should render correctly with no props', () => {
    const wrapper = renderer.create(<MovieCard />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with custom props', () => {
    const wrapper = renderer.create(<MovieCard name={'movie'} year={'2000'} url={'abc'} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
