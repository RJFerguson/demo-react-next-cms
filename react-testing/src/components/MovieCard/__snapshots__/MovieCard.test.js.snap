// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard Component should render correctly with custom props 1`] = `
<div
  className="MovieCard"
>
  <img
    alt="movies"
    src="https://image.tmdb.org/t/p/w200/abc"
  />
  <div>
    <h3>
      movie
    </h3>
    <p>
      2000
    </p>
  </div>
</div>
`;

exports[`MovieCard Component should render correctly with no props 1`] = `
<div
  className="MovieCard"
>
  <img
    alt="movies"
    src="https://image.tmdb.org/t/p/w200/undefined"
  />
  <div>
    <h3 />
    <p />
  </div>
</div>
`;
